src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1966,11 +2006,21 @@ mgp_error mgp_edge_get_type(mgp_edge *e, mgp_edge_type *result) {
 }
 
 mgp_error mgp_edge_get_from(mgp_edge *e, mgp_vertex **result) {
+  if (e->from.graph->ctx && e->from.graph->ctx->auth_checker &&
+      !e->from.graph->ctx->auth_checker->Accept(*e->from.graph->ctx->db_accessor, e->from.impl, e->from.graph->view)) {
+    return mgp_error::MGP_ERROR_NO_ERROR;
+  }
+
   *result = &e->from;
   return mgp_error::MGP_ERROR_NO_ERROR;
 }
 
 mgp_error mgp_edge_get_to(mgp_edge *e, mgp_vertex **result) {
+  if (e->from.graph->ctx && e->from.graph->ctx->auth_checker &&
+      !e->from.graph->ctx->auth_checker->Accept(*e->from.graph->ctx->db_accessor, e->from.impl, e->from.graph->view)) {
+    return mgp_error::MGP_ERROR_NO_ERROR;
+  }
+
```

Comment:
what are you checking here?

Reply:
I was thinking the same. As the edge iterators are checking to have access to the other side of the edge, I am wondering how you pass in an edge for which this will filter out any vertices.

Reply:
Edge Iterators are filtering out edges if user cant see both sides of the edge